custom_components/tahoma/tahoma_api.py

```diff
@@ -593,108 +593,33 @@ def __init__(self, protocol, dataInput):
         self.__protocol = protocol
         self.__raw_data = dataInput
         self.__active_states = {}
-
-        debug_output = json.dumps(dataInput)
-
-        if "label" not in dataInput.keys():
-            raise ValueError("No device name found: " + debug_output)
-
-        self.__label = dataInput["label"]
-
-        if "controllableName" not in dataInput.keys():
-            raise ValueError("No control label name found: " + debug_output)
-
-        self.__type = dataInput["controllableName"]
-
-        if "deviceURL" not in dataInput.keys():
-            raise ValueError("No control URL: " + debug_output)
-
-        self.__url = dataInput["deviceURL"]
-
-        if "uiClass" not in dataInput.keys():
-            raise ValueError("No ui Class: " + debug_output)
-
-        self.__uiclass = dataInput["uiClass"]
-
-        if "widget" not in dataInput.keys():
-            raise ValueError("No widget: " + debug_output)
-
-        self.__widget = dataInput["widget"]
-
+        self.__label = dataInput.get("label", "")
```

Comment:
I would not set them as an empty string, can we set them as None or the default type (not sure, a dict?)

Reply:
I think those 5 ones are strings!

Reply:
`definition` is a bit trickier as it's a dict.